java-desktop/src/main/java/com/couchbase/todo/controller/ConfigController.java

```diff
@@ -0,0 +1,109 @@
+package com.couchbase.todo.controller;
+
+import java.net.URL;
+import java.util.Objects;
+import java.util.ResourceBundle;
+
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Button;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.TextField;
+import javafx.stage.Stage;
+import org.jetbrains.annotations.NotNull;
+
+import com.couchbase.todo.TodoApp;
+import com.couchbase.todo.model.Config;
+import com.couchbase.todo.model.DB;
+
+
+public class ConfigController implements Initializable {
+    private @NotNull Stage stage;
+
+    @FXML
+    private CheckBox loggingCheckbox;
+    @FXML
+    private CheckBox loginCheckbox;
+    @FXML
+    private CheckBox localCheckbox;
+    @FXML
+    private CheckBox remoteCheckbox;
+    @FXML
+    private TextField cblVersionTextField;
+    @FXML
+    private TextField dbNameTextField;
+    @FXML
+    private TextField sgUrlTextField;
+    @FXML
+    private TextField maxRetriesTextField;
+    @FXML
+    private TextField waitTimeTextField;
+    @FXML
+    private Button saveButton;
+    @FXML
+    private Button cancelButton;
+
+    public ConfigController(@NotNull Stage stage) { this.stage = stage; }
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        //set initial values of checkbox and text field
+        buttonsEventHandler();
```

Comment:
Not loving this name.  What does it do? This name suggests that it *is* the button event handler.  I think it will register the button event handler....